pkg/pgmodel/sql_reader.go

```diff
@@ -30,26 +28,16 @@ const (
 	getLabelsSQL       = "SELECT (labels_info($1::int[])).*"
 )
 
-// NewPgxReaderWithMetricCache returns a new DBReader that reads from PostgreSQL using PGX
-// and caches metric table names using the supplied cacher.
-func NewPgxReaderWithMetricCache(c *pgxpool.Pool, cache MetricCache, labelsCacheSize uint64) *DBReader {
+// NewQuerierWithCaches returns a new pgxQuerier that reads from PostgreSQL using PGX
+// and caches metric table names, and label sets using the supplied caches.
+func NewQuerierWithCaches(conn pgxconn.PgxConn, metricCache MetricCache, labelsCache LabelsCache) Querier {
 	pi := &pgxQuerier{
-		conn: &pgxConnImpl{
-			conn: c,
-		},
-		metricTableNames: cache,
-		labels:           clockcache.WithMax(labelsCacheSize),
+		conn:             conn,
+		metricTableNames: metricCache,
+		labels:           labelsCache,
 	}
 
-	return &DBReader{
-		db: pi,
-	}
-}
-
-// NewPgxReader returns a new DBReader that reads that from PostgreSQL using PGX.
-func NewPgxReader(c *pgxpool.Pool, readHist prometheus.ObserverVec, labelsCacheSize uint64) *DBReader {
-	cache := &MetricNameCache{clockcache.WithMax(DefaultMetricCacheSize)}
-	return NewPgxReaderWithMetricCache(c, cache, labelsCacheSize)
+	return pi
```

Comment:
IMO, we should avoid using a var `pi` which does nothing than just holding and simply returning an object.